pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/MLPendingAckReplyCallBack.java

```diff
@@ -44,7 +44,7 @@ public MLPendingAckReplyCallBack(PendingAckHandleImpl pendingAckHandle) {
 
     @Override
     public void replayComplete() {
-        synchronized (pendingAckHandle) {
+        pendingAckHandle.getInternalPinnedExecutor().submit(() -> {
```

Comment:
use execute method. if you use submit, it will return future, we don't need the future